src/types/dropdown-menu.ts

```diff
@@ -14,3 +14,19 @@ export const AppearanceArray = ['menu', 'selectionMenu'] as const
 export type Appearance = AnyElementOf<typeof AppearanceArray>
 
 export type DropdownItemType = 'link' | 'button' | 'default'
+
+export type DropdownItemRenderedType = 'link' | 'router-link' | 'button' | 'default'
+
+export interface IDropdownItemRenderedComponent {
+  tag: string;
+  onClick: ((event: Event) => void) | undefined;
+  attrs: {
+    class: string;
+    isRounded?: boolean;
+    disabled?: boolean;
+    to?: string | object | undefined;
```

Comment:
Since the `to` property is optional, you don't need `undefined` in the type
```suggestion
    to?: string | object
```

Reply:
fixed